internal/cli/federatedauthentication/identityprovider/update/oidc.go

```diff
@@ -0,0 +1,227 @@
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+// atlas federatedAuthentication identityProvider update oidc <identityProviderId> --federationSettingsId federationSettingsId --idpType idpType [--audience audience] [--authorizationType authorizationType] [--clientId clientId] [--description description] [--displayName displayName] [--groupsClaim groupsClaim] [--userClaim userClaim] [--issuerUri issuerUri] [--associatedDomain associatedDomains] [--requestedScope requestedScopes][-o/--output output].
+func OIDCBuilder() *cobra.Command {
+	opts := &OidcOpts{
+		protocol: oidc,
+	}
+	cmd := &cobra.Command{
+		Use:   "oidc [identityProviderId]",
+		Short: "Update an OIDC identity provider.",
+		Args:  cobra.ExactArgs(1),
+		Annotations: map[string]string{
+			"identityProviderIdDesc": "The Identity Provider ID.",
+			"output":                 updateTemplate,
+		},
+		Example: `  # Update the audience for the specified identity provider from your federation settings with ID 5d1113b25a115342acc2d1aa, federationSettingsId 5d1113b25a115342acc2d1aa and IdpType WORKFORCE
+			atlas federatedAuthentication identityProvider update 5d1113b25a115342acc2d1aa --federationSettingsId 5d1113b25a115342acc2d1aa --idpType WORKFORCE --audience newAudience
+		`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			opts.protocol = oidc
+			flags := cmd.Flags()
+			if err := opts.Validate(flags); err != nil {
+				return err
+			}
+			return opts.PreRunE(
+				opts.InitStore(cmd.Context()),
+				opts.InitOutput(cmd.OutOrStdout(), updateTemplate),
+				opts.InitInput(cmd.InOrStdin()),
+			)
+		},
+		RunE: func(_ *cobra.Command, args []string) error {
+			opts.identityProviderID = args[0]
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.federationSettingsID, flag.FederationSettingsID, "", usage.FederationSettingsID)
+	cmd.Flags().StringVar(&opts.idpType, flag.IdpType, "", usage.IdpType)
+	cmd.Flags().StringVar(&opts.audience, flag.Audience, "", usage.Audience)
+	cmd.Flags().StringVar(&opts.authorizationType, flag.AuthorizationType, "", usage.AuthorizationType)
+	cmd.Flags().StringVar(&opts.clientID, flag.ClientID, "", usage.ClientID)
+	cmd.Flags().StringVar(&opts.description, flag.Description, "", usage.Description)
+	cmd.Flags().StringVar(&opts.groupsClaim, flag.GroupsClaim, "", usage.GroupsClaim)
+	cmd.Flags().StringVar(&opts.userClaim, flag.UserClaim, "", usage.UserClaim)
+	cmd.Flags().StringVar(&opts.issuerURI, flag.IssuerURI, "", usage.IssuerURI)
+	cmd.Flags().StringSliceVar(&opts.associatedDomains, flag.AssociatedDomain, []string{}, usage.AssociatedDomains)
+	cmd.Flags().StringSliceVar(&opts.requestedScopes, flag.RequestedScope, []string{}, usage.RequestedScopes)
+
+	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
+	_ = cmd.RegisterFlagCompletionFunc(flag.Output, opts.AutoCompleteOutputFlag())
+	_ = cmd.MarkFlagRequired(flag.FederationSettingsID)
+	_ = cmd.MarkFlagRequired(flag.IdpType)
```

Comment:
@gssbzn in light of our convo last PR, here I still think we should make it required because it should reflect the current value of the IDP that user wants to update, if not provided API throws:

```shell
    "idpType": "Invalid idpType. The idpType must be WORKLOAD, the same as the existing identity provider."
```